src/microbe_stage/OrganelleUpgrades.cs

```diff
@@ -17,6 +18,20 @@ public class OrganelleUpgrades : ICloneable
     /// </summary>
     public IComponentSpecificUpgrades? CustomUpgradeData { get; set; }
 
+    public bool Equals(OrganelleUpgrades other)
+    {
+        if (!UnlockedFeatures.SequenceEqual(other.UnlockedFeatures))
+            return false;
+
+        if (CustomUpgradeData == null)
+            return other.CustomUpgradeData == null;
```

Comment:
So maybe this could have something like `other.CustomUpgradeData == null || other.CustomUpgradeData.EqualsDefault()`?

Reply:
I thought about doing something like that too, but there were a few things I was concerned about.

1. An OrganelleUpgrades for a Chemoreceptor with null CustomUpgradeData would be considered equal to an OrganelleUpgrades for a Lysosome with the default value. Right now, an OrganelleUpgrades for a Chemoreceptor and an OrganelleUpgrades for a Lysosome should never be compared anyway, so this wouldn't cause any problems, it just made me a little worried.
2. LysosomeUpgrades and ChemoreceptorUpgrades don't have any way to know what their default value is, so the logic to figure that out would either have to be moved to those classes or split into a public method on their UpgradeGUIs so they can access it.

Assigning the default value to the upgrade data as soon as the GUI is opened felt a little suspicious to me too, so if you think the problems above are fine, I'll make that change.

Reply:
Those are valid concerns I didn't consider. It would be possible to specify the defaults for those missing upgrade types. But I think maybe it would be easier to just live with the problem of null not being equal to the default upgrades. I don't think it'll be that big of a problem that the user doesn't change any options but just accepts the changes and ends up with a change from null to the default values. At least I see that as the least worst of all the solutions we've come up with.